docs/API.md

```diff
@@ -951,6 +1008,217 @@ __Example__
+    minioClient.enableVersioning("my-bucketname");
+    System.out.println("Object versioning is enabled in my-bucketname.");
+  }
+} catch (MinioException e) {
+  System.out.println("Error occurred: " + e);
+}
+```
+
+
+<a name="disableVersioning"></a>
+### disableVersioning(String bucketName)
+`public void disableVersioning(String bucketName)`
+
+Object versioning is disabled in bucketName.
+
+[View Javadoc](http://minio.github.io/minio-java/io/minio/MinioClient.html#disableVersioning-java.lang.String-)
+
+__Parameters__
+
+|Param   | Type	  | Description  |
+|:--- |:--- |:--- |
+| ``bucketName``  | _String_ | Name of the bucket.  |
+
+
+| Return Type	  | Exceptions	  |
+|:--- |:--- |
+| ``None``  | Listed Exceptions: |
+|        |  ``InvalidBucketNameException`` : upon invalid bucket name. |
+|        |  ``RegionConflictException`` : upon  passed region conflicts with the one previously specified. |
+|        | ``NoSuchAlgorithmException`` : upon requested algorithm was not found during signature calculation.           |
+|        | ``InsufficientDataException`` : Thrown to indicate that reading given InputStream gets EOFException before reading given length. |
+|        | ``IOException`` : upon connection error.            |
+|        | ``InvalidKeyException`` : upon an invalid access key or secret key.           |
+|        | ``NoResponseException`` : upon no response from server.            |
+|        | ``org.xmlpull.v1.XmlPullParserException`` : upon parsing response XML.            |
+|        | ``ErrorResponseException`` : upon unsuccessful execution.            |
+|        | ``InternalException`` : upon internal library error.        |
+|        | ``InvalidResponseException`` : upon a non-xml response from server.        |
+
+__Example__
+
+
+```java
+try {
+    // Object versioning is disabled in my-bucketname
+    minioClient.disableVersioning("my-bucketname");
+    System.out.println("Object versioning is disabled in my-bucketname.");
+  }
+} catch (MinioException e) {
+  System.out.println("Error occurred: " + e);
+}
+```
+
+<a name="setDefaultRetention"></a>
+### setDefaultRetention(String bucketName, ObjectLockConfiguration config)
+`public void setDefaultRetention(String bucketName, ObjectLockConfiguration config)`
+
+Set default retention on bucket.
+
+[View Javadoc](http://minio.github.io/minio-java/io/minio/MinioClient.html#setDefaultRetention-java.lang.String-io.minio.messages.ObjectLockConfiguration- )
+
+__Parameters__
+
+|Param   | Type	  | Description  |
+|:--- |:--- |:--- |
+| ``bucketName``  | _String_ | Name of the bucket.  |
+| ``config``  | _ObjectLockConfiguration_ | Object lock Configuration  |
+
+
+| Return Type	  | Exceptions	  |
+|:--- |:--- |
+| ``None``  | Listed Exceptions: |
+|        |  ``InvalidBucketNameException`` : upon invalid bucket name. |
+|        |  ``RegionConflictException`` : upon  passed region conflicts with the one previously specified. |
```

Comment:
Instead of `RegionConflictException` you should talk about `InvalidArgumentException` as that is what is being thrown in the code.